test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_default_config_test.clj

```diff
@@ -71,19 +75,110 @@ react accordingly."
+        ;; Using reflection here because the .getQueue method is protected and I
+        ;; didn't see any other way to pull the queue back from the thread pool.
+        get-queue-method (-> thread-pool
+                             (.getClass)
+                             (.getDeclaredMethod "getQueue" nil))
+        _                (.setAccessible get-queue-method true)]
+    (.invoke get-queue-method thread-pool nil)))
 
 (deftest default-server-settings-test
   (let [server (Server.)]
-    (is (= 30000 (.getStopTimeout server)))))
-
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/component/AbstractLifeCycle.java#L48
+    (is (= 30000 (.getStopTimeout server))
+        "Unexpected default for 'shutdown-timeout-seconds'")
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#L67
+    (is (= 200 (get-max-threads-for-server server))
+        "Unexpected default for 'max-threads'")
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java#L117
+    (is (= (Integer/MAX_VALUE) (.getMaxCapacity
+                                 (get-server-thread-pool-queue server)))
+        "Unexpected default for 'queue-max-size'")))
 
+(def threads-per-connector
+  "The total number of threads needed per attached connector."
+  (+ acceptor-thread-count selector-thread-count))
 
+(defn calculate-minimum-required-threads
+  "Calculate the minimum number threads that a single Jetty Server instance
+  needs.  See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java#L334-L350"
+  [connectors]
+  (+ 1 (* connectors threads-per-connector)))
 
+(deftest default-min-threads-settings-test
+  ;; This test just exists to validate the advice we give for the bare
+  ;; minimum number of threads that one should account for when setting the
+  ;; 'max-threads' setting for a server instance.
+  (letfn [(get-server [max-threads connectors]
+            (let [server (Server. (QueuedThreadPool. max-threads))]
+              (dotimes [_ connectors]
+                (.addConnector server (ServerConnector. server)))
+              server))
+          (insufficient-threads-msg [server]
+            (let [connectors (count (.getConnectors server))]
+              (re-pattern (str "Insufficient threads: max="
+                               (get-max-threads-for-server server)
+                               " < needed\\(acceptors="
+                               (* acceptor-thread-count connectors)
+                               " \\+ selectors="
+                               (* selector-thread-count connectors)
+                               " \\+ request=1\\)"))))]
+    (dotimes [x 5]
```

Comment:
It's arbitrary.  Could just do 2 times to cover the maximum number we'd ever use for a connector, for now.  Our testing here is indirect anyway since we're throwing some numbers and Jetty and using our expectation of what it will calculate to give us some confidence that we know how their algorithm works.  I'd thought that by throwing a few more connectors in there that it might increase that confidence a bit more.  I could be convinced to bump it back down to 2, though, if you think that's better.